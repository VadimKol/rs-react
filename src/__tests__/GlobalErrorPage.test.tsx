import { render } from '@testing-library/react';

import GlobalErrorPage from '@/app/global-error';

describe('Global error page', () => {
  const c = console;
  c.error = jest.fn();

  it('renders correctly', () => {
    const { container } = render(<GlobalErrorPage error={new Error('Some error')} />);

    expect(container).toMatchSnapshot();
  });
});
