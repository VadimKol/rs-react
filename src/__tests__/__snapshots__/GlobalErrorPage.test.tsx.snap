// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Global error page renders correctly 1`] = `
<div>
  <html
    lang="en"
  >
    <body
      class="body"
    >
      <main
        class="main"
      >
        <div
          class="error"
        >
          <p
            class="name"
          >
            ERROR
          </p>
          <img
            alt="Rick and Morty error"
            class="error_img"
            data-nimg="1"
            decoding="async"
            height="500"
            loading="lazy"
            src="/_next/image?url=%2Fimg.jpg&w=1080&q=75"
            srcset="/_next/image?url=%2Fimg.jpg&w=640&q=75 1x, /_next/image?url=%2Fimg.jpg&w=1080&q=75 2x"
            style="color: transparent;"
            width="500"
          />
          <p
            class="desc"
          >
            Some error
          </p>
          <p
            class="desc"
          >
            Please refresh the page
          </p>
        </div>
      </main>
    </body>
  </html>
</div>
`;
